core/src/graph/mod.rs

```diff
@@ -191,6 +197,276 @@ derive_from! {
+                }
+            }
+            Value::Object(_) => {
+                <serde_json::Value as Encode<'_, Postgres>>::encode(serde_json::to_value(self).unwrap(), buf)
+            },
+            Value::Array(arr) => arr.encode_by_ref(buf), //TODO special-case for BYTEA
+        }
+    }
+}
+
+impl Encode<'_, MySql> for Value {
+    fn encode_by_ref(
+        &self,
+        buf: &mut Vec<u8>
+    ) -> IsNull {
+        match self {
+            Value::Null(_) => IsNull::Yes,
+            Value::Bool(b) => <bool as Encode<'_, MySql>>::encode_by_ref(b, buf),
+            Value::Number(num) => {
+                match *num {
+                    Number::I8(i) => <i8 as Encode<'_, MySql>>::encode_by_ref(&i, buf),
+                    Number::I16(i) => <i16 as Encode<'_, MySql>>::encode_by_ref(&i, buf),
+                    Number::I32(i) => <i32 as Encode<'_, MySql>>::encode_by_ref(&i, buf),
+                    Number::I64(i) => <i64 as Encode<'_, MySql>>::encode_by_ref(&i, buf),
+                    Number::I128(i) => <sqlx::types::Decimal as Encode<'_, MySql>>::encode_by_ref(&i.into(), buf),
+                    Number::U8(i) => <i8 as Encode<'_, MySql>>::encode_by_ref(&(i as i8), buf),
+                    Number::U16(i) => <i16 as Encode<'_, MySql>>::encode_by_ref(&(i as i16), buf),
+                    Number::U32(i) => <u32 as Encode<'_, MySql>>::encode_by_ref(&i, buf),
+                    Number::U64(i) => <i64 as Encode<'_, MySql>>::encode_by_ref(&(i as i64), buf),
+                    Number::U128(i) => <sqlx::types::Decimal as Encode<'_, MySql>>::encode_by_ref(&i.into(), buf),
+                    Number::F32(f) => <f32 as Encode<'_, MySql>>::encode_by_ref(&f, buf),
+                    Number::F64(f) => <f64 as Encode<'_, MySql>>::encode_by_ref(&f, buf),
+                }
+            },
+            Value::String(s) => <String as Encode<'_, MySql>>::encode_by_ref(s, buf),
+            Value::DateTime(dt) => {
+                match dt {
+                    ChronoValue::NaiveDate(nd) => <NaiveDate as Encode<'_, MySql>>::encode_by_ref(nd, buf),
+                    ChronoValue::NaiveTime(nt) => <NaiveTime as Encode<'_, MySql>>::encode_by_ref(nt, buf),
+                    ChronoValue::NaiveDateTime(ndt) => <NaiveDateTime as Encode<'_, MySql>>::encode_by_ref(ndt, buf),
+                    ChronoValue::DateTime(dt) => <DateTime<Utc> as Encode<'_, MySql>>::encode_by_ref(&dt.with_timezone(&Utc), buf),
+                }
+            }
+            Value::Object(_) => {
+                <serde_json::Value as Encode<'_, MySql>>::encode(serde_json::to_value(self).unwrap(), buf)
+            },
+            Value::Array(_arr) => todo!()//<Vec<Value> as Encode<'_, MySql>>::encode_by_ref(arr, buf), //TODO special-case for u8 arrays?
+        }
+    }
+
+    fn produces(&self) -> Option<MySqlTypeInfo> {
+        Some(match self {
```

Comment:
I initially had that, but because of the `Db` generic, inference won't allow us to use method call syntax here, so we just end up with even more code. So we may as well use `type_info` here.